modules/e2ee/OlmAdapter.js

```diff
@@ -62,39 +62,58 @@ export class OlmAdapter extends Listenable {
         this._key = undefined;
         this._keyIndex = -1;
         this._reqs = new Map();
+        this._sessionInitialization = undefined;
 
         if (OlmAdapter.isSupported()) {
             this._bootstrapOlm();
 
             this._conf.on(JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED, this._onEndpointMessageReceived.bind(this));
-            this._conf.on(JitsiConferenceEvents.CONFERENCE_JOINED, this._onConferenceJoined.bind(this));
             this._conf.on(JitsiConferenceEvents.CONFERENCE_LEFT, this._onConferenceLeft.bind(this));
             this._conf.on(JitsiConferenceEvents.USER_LEFT, this._onParticipantLeft.bind(this));
+            this._conf.on(JitsiConferenceEvents.PARTICIPANT_PROPERTY_CHANGED,
+                this._onParticipantPropertyChanged.bind(this));
         } else {
             this._init.reject(new Error('Olm not supported'));
         }
     }
 
     /**
-     * Indicates if olm is supported on the current platform.
-     *
-     * @returns {boolean}
+     * Starts new olm sessions with every other participant that has the participantId "smaller" the localParticipantId.
      */
-    static isSupported() {
-        return typeof window.Olm !== 'undefined';
+    async initSessions() {
+        if (this._sessionInitialization) {
```

Comment:
Replace this check with a throw / assert since it should be impossible for this to happen, and if it does it's a programming error, not a runtime error.